packages/unenv-preset/build.config.ts

```diff
@@ -1,19 +0,0 @@
-import { defineBuildConfig } from "unbuild";
-
-export default defineBuildConfig({
-	declaration: true,
-	rollup: {
-		emitCJS: true,
-	},
-	entries: [
-		"src/index",
-		{ input: "src/runtime/", outDir: "dist/runtime", format: "esm" },
-		{
-			input: "src/runtime/",
-			outDir: "dist/runtime",
-			format: "cjs",
-			ext: "cjs",
-			declaration: false,
```

Comment:
This line was actually instructing unbuild to emit CJS (it is not a default behavior).

Using tsc makes sense for consistency with cloudflare repo. unbuild uses mkdist and transforms such as ensuring output relative imports are ESM-compatible (with explicit `.mjs` extension). The behavior might vary when using tsc alone..

Reply:
> The behavior might vary when using tsc alone and less consistent with unenv artifacts (which use unbuild/mkdist for ESM compliance)

Do you foresee any issues?

I switched to `tsc` for 2 reasons:

- consistency with other packages in `workers-sdk`
- I had troubles compiling tests to a different `outDir` with `unbuild`

Reply:
> Consistency with other packages in workers-sdk

consistency with unenv output and (end-user) that consume the dist I believe is more important for this case. strict ESM compliance i think is also an important point for us.

> I had troubles compiling tests to a different outDir with unbuild

Do you have a branch that was trying to do this? (also why tests need compilation?)

Reply:
> also why tests need compilation?

We need to launch `wrangler deploy --dry-run` with a custom resolution path (the current package) and a custom `workerd` (from the current package), both are set by env variables. So what we need to compile if the file that will launch this build.

> Do you have a branch that was trying to do this?

Basically what I had was:

```
	entries: [
		"src/index",
		{ input: "src/runtime/", outDir: "dist/runtime", format: "esm" },
		{ input: "tests/build.ts", outDir: "dist-test", format: "esm" },
	]
```    

From the log, I could see that `dist-test/` was cleared but still build.ts would be compiled into `dist/tests/build.esm`

Reply:
(haven't locally tested) you might want to set `builder: 'mkdist'` for second entry. When there is traling slash like `runtime/` it is infered (default is rollup build with multi entries).

Alternatively you could tsc only to emit test artifacts..